static/app/views/automations/components/automationHistoryList.tsx

```diff
@@ -1,55 +1,158 @@
-      <SimpleTable.Header>
-        <SimpleTable.HeaderCell>
-          {tct('Time Sent ([timezone])', {timezone: moment.tz(timezone).zoneAbbr()})}
-        </SimpleTable.HeaderCell>
-        <SimpleTable.HeaderCell>Monitor</SimpleTable.HeaderCell>
-        <SimpleTable.HeaderCell>Issue</SimpleTable.HeaderCell>
-      </SimpleTable.Header>
-      {history.length === 0 && <SimpleTable.Empty>No history found</SimpleTable.Empty>}
-      {history.map((row, index) => (
+    <Fragment>
+      {Array.from({length: DEFAULT_HISTORY_PER_PAGE}).map((_, index) => (
         <SimpleTable.Row key={index}>
           <SimpleTable.RowCell>
-            <DateTime date={row.dateSent} forcedTimezone={timezone} />
+            <Placeholder height="20px" />
           </SimpleTable.RowCell>
-          <SimpleTable.RowCell>
-            <Link to={row.monitor.link}>{row.monitor.name}</Link>
+          <SimpleTable.RowCell data-column-name="type">
+            <Placeholder height="20px" />
           </SimpleTable.RowCell>
-          <SimpleTable.RowCell>
-            <Link to={`/issues/${row.groupId}`}>{`#${row.groupId}`}</Link>
+          <SimpleTable.RowCell data-column-name="last-issue">
+            <Placeholder height="20px" />
+          </SimpleTable.RowCell>
+          <SimpleTable.RowCell data-column-name="owner">
+            <Placeholder height="20px" />
           </SimpleTable.RowCell>
         </SimpleTable.Row>
       ))}
-    </SimpleTableWithColumns>
+    </Fragment>
+  );
+}
+
+export default function AutomationHistoryList({
+  automationId,
+  limit = DEFAULT_HISTORY_PER_PAGE,
+  query,
+  emptyMessage = t('No history found'),
+}: Props) {
+  const org = useOrganization();
+  const timezone = useTimezone();
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const cursor =
+    typeof location.query.cursor === 'string' ? location.query.cursor : undefined;
+
+  const {
+    data: fireHistory = [],
+    isLoading,
+    isError,
+    getResponseHeader,
+  } = useAutomationFireHistoryQuery(
+    {id: automationId, limit, cursor, query},
+    {enabled: !!automationId}
+  );
+
+  const pageLinks = getResponseHeader?.('Link');
+
+  return (
+    <Fragment>
+      <SimpleTableWithColumns>
+        <SimpleTable.Header>
+          <SimpleTable.HeaderCell>
+            {tct('Time Sent ([timezone])', {timezone: moment.tz(timezone).zoneAbbr()})}
+          </SimpleTable.HeaderCell>
+          <SimpleTable.HeaderCell>{t('Monitor')}</SimpleTable.HeaderCell>
+          <SimpleTable.HeaderCell>{t('Issue')}</SimpleTable.HeaderCell>
+          <SimpleTable.HeaderCell>{t('Alerts')}</SimpleTable.HeaderCell>
+        </SimpleTable.Header>
+        {isLoading && <Skeletons />}
+        {isError && <LoadingError />}
+        {!isLoading && !isError && fireHistory.length === 0 && (
+          <SimpleTable.Empty>{emptyMessage}</SimpleTable.Empty>
+        )}
+        {fireHistory.map((row, index) => (
+          <SimpleTable.Row key={index}>
```

Comment:
each fire probably has a unique id?
```suggestion
        {fireHistory.map((row) => (
          <SimpleTable.Row key={row.id}>
```

Reply:
the current api response doesn't look like it technically has to have anything unique, based on the model, but i could update the response to include a unique id if that's preferred

https://github.com/getsentry/sentry/blob/7c9291261cb6b178b3b0bde2ce6d8f01a1769d29/src/sentry/workflow_engine/endpoints/serializers.py#L466-L471

https://github.com/getsentry/sentry/blob/3503fdd1cf865d749c5faabcf8de3d6e1c34e44a/src/sentry/workflow_engine/models/workflow_fire_history.py#L11-L19